lib/index.js

```diff
@@ -56,6 +56,11 @@ class UI {
     this.writeLevel = options.writeLevel || DEFAULT_WRITE_LEVEL;
     this.errorReport = null;
     this.ci = !!options.ci;
+    
+    // Ensure spinner is stopped on process termination
+    process.on('SIGINT', () => {
```

Comment:
I think we want to ensure we've stopped progress whenever the process is going to exit, not just when it exits for `SIGINT`.

```suggestion
    process.on('exit', () => {
```